spanner/src/update_backup.php

```diff
@@ -40,10 +41,14 @@ function update_backup($instanceId, $backupId)
     $spanner = new SpannerClient();
     $instance = $spanner->instance($instanceId);
     $backup = $instance->backup($backupId);
+    $backup->reload();
 
-    // Expire time must be within 366 days of the create time of the backup.
-    $newTimestamp = new \DateTime('+30 days');
-    $backup->updateExpireTime($newTimestamp);
+    // The new expire time can't be greater than maxExpireTime for the backup.
+    $newExpireTime = new DateTime('+30 days');
+    $maxExpireTime = new DateTime($backup->info()['maxExpireTime']);
+    $newExpireTime = $newExpireTime > $maxExpireTime ? $maxExpireTime : $newExpireTime;
+
+    $backup->updateExpireTime($newExpireTime);
```

Comment:
The php [max function](https://www.php.net/manual/en/function.max.php) works with datetimes as well:
```suggestion
    $backup->updateExpireTime(max($newExpireTime, $maxExpireTime));
```

Reply:
Used the suggestion, but just to clarify, in this case we need `min`.